python/mxnet/optimizer.py

```diff
@@ -143,8 +143,8 @@ def _update_count(self, index):
         """
         if index not in self._index_update_count:
             self._index_update_count[index] = self.begin_num_update
-        self._index_update_count[index] += 1
-        self.num_update = max(self._index_update_count[index], self.num_update)
```

Comment:
looks good to me, since `_index_update_count` is intialized with `begin_num_update`, we can drop the `max` here.